src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1172,6 +1173,54 @@
   (jdbc/do-commands
     "DROP INDEX IF EXISTS resource_events_resource_type_idx"))
 
+(defn merge-fact-values-into-facts
+  []
+  (jdbc/do-commands
+   ["create table facts_transform"
+    "  (factset_id bigint not null,"
+    "   fact_path_id bigint not null,"
+    "   value_integer bigint,"
+    "   value_float double precision,"
+    "   value_type_id integer not null,"
+    "   value_string text,"
+    "   value jsonb,"
+    "   large_value_hash bytea,"
+    "   value_boolean boolean)"]
+
+   ["insert into facts_transform"
+    "  (factset_id,"
+    "   fact_path_id,"
+    "   value_integer,"
+    "   value_float,"
+    "   value_type_id,"
+    "   value_string,"
+    "   value,"
+    "   large_value_hash,"
+    "   value_boolean)"
+    "  select factset_id, fact_path_id,"
+    "         value_integer, value_float, value_type_id, value_string, value,"
+    "         case when pg_column_size(value) >= 50"
+    "              then value_hash"
+    "         end,"
+    "         value_boolean"
+    "    from facts inner join fact_values on fact_value_id = fact_values.id"]
+
+   "drop index facts_fact_value_id_idx"
```

Comment:
Ahh, I think that's just an artifact of the vector for the multi-line command above (newly allowed by https://github.com/puppetlabs/puppetdb/pull/2164/commits/14f6c75dc9900160e3f4f3680e1048ada86b5b3a in this series), which might also be a thing to discuss.